stake-pool/js/src/index.ts

```diff
@@ -505,3 +578,331 @@ export async function withdrawSol(
+ */
+export async function updateStakePool(
+  connection: Connection,
+  stakePool: StakePoolAccount,
+  noMerge = false,
+) {
+  // const stakePool = await getStakePoolAccount(connection, stakePoolAddress)
+  const stakePoolAddress = stakePool.pubkey;
+
+  const validatorList = await getValidatorListAccount(
+    connection,
+    stakePool.account.data.validatorList,
+  );
+
+  const withdrawAuthority = await findWithdrawAuthorityProgramAddress(
+    STAKE_POOL_PROGRAM_ID,
+    stakePoolAddress,
+  );
+
+  const updateListInstructions: TransactionInstruction[] = [];
+  const instructions: TransactionInstruction[] = [];
+
+  let startIndex = 0;
+  const validatorChunks: Array<ValidatorStakeInfo[]> = arrayChunk(
+    validatorList.account.data.validators,
+    MAX_VALIDATORS_TO_UPDATE,
+  );
+
+  for (const validatorChunk of validatorChunks) {
+    const validatorAndTransientStakePairs: PublicKey[] = [];
+
+    for (const validator of validatorChunk) {
+      const validatorStake = await findStakeProgramAddress(
+        STAKE_POOL_PROGRAM_ID,
+        validator.voteAccountAddress,
+        stakePoolAddress,
+      );
+      validatorAndTransientStakePairs.push(validatorStake);
+
+      const transientStake = await findTransientStakeProgramAddress(
+        STAKE_POOL_PROGRAM_ID,
+        validator.voteAccountAddress,
+        stakePoolAddress,
+        validator.transientSeedSuffixStart,
+      );
+      validatorAndTransientStakePairs.push(transientStake);
+    }
+
+    updateListInstructions.push(
+      StakePoolInstruction.updateValidatorListBalance({
+        stakePool: stakePoolAddress,
+        validatorList: stakePool.account.data.validatorList,
+        reserveStake: stakePool.account.data.reserveStake,
+        validatorAndTransientStakePairs,
+        withdrawAuthority,
+        startIndex,
+        noMerge,
+      }),
+    );
+    startIndex += MAX_VALIDATORS_TO_UPDATE;
+  }
+
+  instructions.push(
+    StakePoolInstruction.updateStakePoolBalance({
+      stakePool: stakePoolAddress,
+      validatorList: stakePool.account.data.validatorList,
+      reserveStake: stakePool.account.data.reserveStake,
+      managerFeeAccount: stakePool.account.data.managerFeeAccount,
+      poolMint: stakePool.account.data.poolMint,
+      withdrawAuthority,
+    }),
+  );
+
+  instructions.push(
+    StakePoolInstruction.cleanupRemovedValidatorEntries({
+      stakePool: stakePoolAddress,
+      validatorList: stakePool.account.data.validatorList,
+    }),
+  );
+
+  return {
+    updateListInstructions,
+    finalInstructions: instructions,
+  };
+}
+
+/**
+ * Retrieves detailed information about the StakePool.
+ */
+export async function stakePoolInfo(connection: Connection, stakePoolAddress: PublicKey) {
+  const stakePool = await getStakePoolAccount(connection, stakePoolAddress);
+  const reserveAccountStakeAddress = stakePool.account.data.reserveStake;
+  const totalLamports = stakePool.account.lamports;
```

Comment:
The total lamports should be the total lamports of the pool, and not the stake pool account, so something like `stakePool.account.data.totalLamports`